pkg/cmd/cli/cmd/create_route.go

```diff
@@ -0,0 +1,396 @@
+	cmd.Flags().String("expose", "", "Name of the service that the new route is exposing")
+	cmd.MarkFlagRequired("expose")
+	cmd.Flags().String("path", "", "Path that the router watches to route traffic to the service.")
+	cmd.Flags().String("cert", "", "Path to a certificate file.")
+	cmd.Flags().String("key", "", "Path to a key file.")
+	cmd.Flags().String("ca-cert", "", "Path to a CA certificate file.")
+	cmd.Flags().String("dest-ca-cert", "", "Path to a CA certificate file, used for securing the connection from the router to the destination.")
+	cmd.MarkFlagRequired("dest-ca-cert")
+
+	return cmd
+}
+
+// CreateReencryptRoute implements the behavior to run the create reencrypt route command.
+func CreateReencryptRoute(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	oc, kc, err := f.Clients()
+	if err != nil {
+		return err
+	}
+
+	ns, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	route, err := unsecuredRoute(cmd, kc, ns, args)
+	if err != nil {
+		return err
+	}
+
+	route.Spec.Host = kcmdutil.GetFlagString(cmd, "hostname")
+	route.Spec.Path = kcmdutil.GetFlagString(cmd, "path")
+
+	route.Spec.TLS = new(api.TLSConfig)
+	route.Spec.TLS.Termination = api.TLSTerminationEdge
+
+	cert, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "cert-file"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.Certificate = string(cert)
+	key, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "key-file"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.Key = string(key)
+	caCert, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "ca-cert-file"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.CACertificate = string(caCert)
+	destCACert, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "dest-ca-cert"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.DestinationCACertificate = string(destCACert)
+
+	route, err = oc.Routes(ns).Create(route)
+	if err != nil {
+		return err
+	}
+	mapper, typer := f.Object()
+	resourceMapper := &resource.Mapper{
+		ObjectTyper:  typer,
+		RESTMapper:   mapper,
+		ClientMapper: resource.ClientMapperFunc(f.Factory.RESTClient),
+	}
+	info, err := resourceMapper.InfoForObject(route)
+	if err != nil {
+		return err
+	}
+	shortOutput := kcmdutil.GetFlagString(cmd, "output") == "name"
+	kcmdutil.PrintSuccess(mapper, shortOutput, out, info.Mapping.Resource, info.Name, "created")
+	return nil
+}
+
+// unsecuredRoute will return a route with enough info so that it can direct traffic to
+// the service provided by --expose. Callers of this helper are responsible for providing
+// tls configuration, path, and the hostname of the route.
+func unsecuredRoute(cmd *cobra.Command, kc *kclient.Client, namespace string, args []string) (*api.Route, error) {
+	serviceName := kcmdutil.GetFlagString(cmd, "expose")
+	// Right now the service is required to exist in order for oc create route
```

Comment:
> We need the service to support tcp in order to expose it. Also it's useful for re-using labels and introspecting the ports in order to set the route port if the user hasn't specified one

Ok, I don't feel a need to rock the boat too badly here, but I don't like false dependencies.  I could very easily create a fake service to expose a route, then delete it and create a new resource that will be routed to without any of the "matching" data lifted.